packages/block-library/src/navigation/edit/index.js

```diff
@@ -336,53 +334,30 @@ function Navigation( {
 		}
 	}, [ clientId, ref, hasUncontrolledInnerBlocks, controlledInnerBlocks ] );
 
-	useEffect( () => {
-		const setPermissionsNotice = () => {
-			if ( noticeRef.current ) {
-				return;
-			}
-
-			noticeRef.current =
-				'block-library/core/navigation/permissions/update';
-
-			createWarningNotice(
-				__(
-					'You do not have permission to edit this Menu. Any changes made will not be saved.'
-				),
-				{
-					id: noticeRef.current,
-					type: 'snackbar',
-				}
-			);
-		};
-
-		const removePermissionsNotice = () => {
-			if ( ! noticeRef.current ) {
-				return;
-			}
-			removeNotice( noticeRef.current );
-			noticeRef.current = null;
-		};
-
-		if ( ! isSelected && ! isInnerBlockSelected ) {
-			removePermissionsNotice();
-		}
-
-		if (
+	useNavigationNotice( {
```

Comment:
My two notes:
* This naming and structure is very specific to navigation, but the implementation is generic
* I had to check how `useNavigationNotice` is implemented to reason about this component. Can we make it obvious? Below is an idea

```js
const [ showCantEditNotice, hideCantEditNotice ] = useNotice( /*...*/ );
const [ showCantCreateNotice, hideCantCreateNotice ] = useNotice( /*...*/ );
useEffect(() => {
    if( ! isSelected && ! isInnerBlockSelected ) {
        hideCantEditNotice();
        hideCantCreateNotice();
    } else if ( ref ) {
        hideCantCreateNotice();
    }

    if (
       ( isSelected || isInnerBlockSelected ) &&
	hasResolvedCanUserUpdateNavigationEntity &&
    ) {
        if ( ! canUserUpdateNavigationEntity ) {
            showCantEditNotice();
        } else if ( ! canUserCreateNavigation ) {
            showCantCreateNotice();
        }
    }
});
```

Reply:
Yeh I didn't like the implementation of those hooks. Just rushing really to land the fix. Let me try and refactor...

Reply:
Refactored ✅